models/repo/pushmirror.go

```diff
@@ -87,10 +95,26 @@ func GetPushMirrorByID(ID int64) (*PushMirror, error) {
 	return m, nil
 }
 
+// GetPushMirrorByRepoIDAndName returns push-mirror information.
+func GetPushMirrorByRepoIDAndName(repoID int64, remoteName string) (*PushMirror, error) {
+	m := &PushMirror{}
+	has, err := db.GetEngine(db.DefaultContext).Where("repo_id = ? AND remote_name = ?", repoID, remoteName).Get(m)
+	if err != nil {
+		return nil, err
+	} else if !has {
+		return nil, ErrPushMirrorNotExist
+	}
+	return m, nil
+}
+
 // GetPushMirrorsByRepoID returns push-mirror information of a repository.
-func GetPushMirrorsByRepoID(repoID int64) ([]*PushMirror, error) {
+func GetPushMirrorsByRepoID(repoID int64, listOptions db.ListOptions) ([]*PushMirror, error) {
```

Comment:
As above.

Reply:
Please remove the other functions now.

Reply:
To summarize, currently we have 2 functions : 

-  GetPushMirrors : to  get a push mirror using a repoID and (pushMirrror ID or remoteName)
-  GetPushMirrorsByRepoID : this returns a **list** of push mirrors belonging to the repo, I kept this function cause I think  it's better to do not mix between get a list of push mirrors and get only one push mirror. 

Should we merge also this one ? if soo all the calling functions should parse the response as a list and I don't think it's a good idea

Reply:
Woops I'd missed that.